pinot-common/src/main/java/com/linkedin/pinot/common/utils/ServiceStatus.java

```diff
@@ -63,6 +71,106 @@ public static Status getServiceStatus() {
     }
   }
 
+  public static class IdealStateAndCurrentStateMatchServiceStatusCallback implements ServiceStatusCallback {
+
+    private String _clusterName;
+    private String _instanceName;
+    private List<String> _resourcesToMonitor;
+    private boolean _finishedStartingUp = false;
+    private HelixManager _helixManager;
+    private HelixAdmin _helixAdmin;
+    private HelixDataAccessor _helixDataAccessor;
+    private Builder _keyBuilder;
+
+    public IdealStateAndCurrentStateMatchServiceStatusCallback(HelixManager helixManager, String clusterName, String instanceName) {
+      _helixManager = helixManager;
+      _clusterName = clusterName;
+      _instanceName = instanceName;
+      _helixAdmin = _helixManager.getClusterManagmentTool();
+      _helixDataAccessor = _helixManager.getHelixDataAccessor();
+      _keyBuilder = _helixDataAccessor.keyBuilder();
+      // Make a list of the resources to monitor
+      _resourcesToMonitor = new ArrayList<>();
+
+      for (String resource : _helixAdmin.getResourcesInCluster(_clusterName)) {
+        final IdealState idealState = _helixAdmin.getResourceIdealState(_clusterName, resource);
+        for (String partitionInResource : idealState.getPartitionSet()) {
+          if (idealState.getInstanceSet(partitionInResource).contains(_instanceName)) {
+            _resourcesToMonitor.add(resource);
+            break;
+          }
+        }
+      }
+      LOGGER.info("Monitoring resources {} for start up of instance {}", _resourcesToMonitor, _instanceName);
+    }
+
+    public IdealStateAndCurrentStateMatchServiceStatusCallback(HelixManager helixManager, String clusterName, String instanceName,
+        List<String> resourcesToMonitor) {
+      _helixManager = helixManager;
+      _clusterName = clusterName;
+      _instanceName = instanceName;
+      _helixAdmin = _helixManager.getClusterManagmentTool();
+      _resourcesToMonitor = resourcesToMonitor;
+
+      LOGGER.info("Monitoring resources {} for start up of instance {}", _resourcesToMonitor, _instanceName);
```

Comment:
if there are 100s of tables hosting partitions on a server, will this log all the table names? Would it be better to log this as debug, or just log the number of reosurces that need to be monitored?